java-lib/src/main/java/com/wavefront/ingester/SourceTagIngesterFormatter.java

```diff
@@ -0,0 +1,324 @@
+package com.wavefront.ingester;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+
+import org.antlr.v4.runtime.ANTLRInputStream;
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.Lexer;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+import org.antlr.v4.runtime.Token;
+
+import java.util.ArrayDeque;
+import java.util.List;
+import java.util.Queue;
+import java.util.logging.Logger;
+
+import sunnylabs.report.ReportSourceTag;
+import queryserver.parser.DSWrapperLexer;
+
+/**
+ * This class can be used to parse sourceTags and description.
+ *
+ * @author Suranjan Pramanik (suranjan@wavefront.com).
+ */
+public class SourceTagIngesterFormatter {
```

Comment:
Do you need to create another ingester formmater for this? Can you not reuse the same formatter class instead of copying a lot of code over?